statsdbinterface/database/models.py

```diff
@@ -178,58 +199,29 @@ def re(self):
         return redeclipse.versions.get_game_version(self.id)
 
     def is_timed(self):
-        return (self.mode == self.re().modes['race'] and
-                'timed' in self.re().mutslist(self.mode, self.mutators))
+        return (self.mode.name == 'race' and
+                'timed' in set(m.name for m in self.mutators))
 
     def is_peaceful(self):
-        return (self.mode == self.re().modes['race'] and
-                'gauntlet' not in self.re().mutslist(self.mode, self.mutators))
+        return (self.mode.name == 'race' and
+                'gauntlet' not in set(m.name for m in self.mutators))
 
     def mode_str(self, short=False):
-        return (self.re().cmodestr[self.mode] if short else
-                self.re().modestr[self.mode])
+        return self.mode.name if short else self.mode.longname
 
     def mutator_list(self, maxlong=0):
-        return self.re().mutslist(self.mode, self.mutators,
-                                  (maxlong and len(self.re().mutslist(
-                                   self.mode, self.mutators)) > maxlong))
-
-    def mutator_dict_list(self, maxlong=0):
-        ret = []
-        re = self.re()
-
-        for m in re.basemuts:
-            if self.mutators & re.basemuts[m]:
-                ret.append({
-                    "link": m,
-                    "name": m,
-                    "longname": m,
-                    "shortname": re.shortmutators[m],
-                    })
-
-        if self.mode in re.gspmuts:
-            for m in re.gspmuts[self.mode]:
-                if self.mutators & re.gspmuts[self.mode][m]:
-                    mn = re.cmodestr[self.mode] + '-' + m
-                    ret.append({
-                        "link": mn,
-                        "name": m,
-                        "longname": m,
-                        "shortname": re.shortmutators[m],
-                    })
-
-        if maxlong and len(ret) > maxlong:
-            for m in ret:
-                m["name"] = m["shortname"]
-        return ret
+        name = 'shortname' if len(self.mutators) > maxlong else 'name'
+        return [getattr(m, name) for m in self.mutators]
```

Comment:
You should return an iterator, that is much faster. If you really needed a list later, then you could still convert it to a list, as calling `list()` on an iterator is as fast as this construct.